### PR TITLE
Users/beedle cmyk/linkedlists

### PR DESCRIPTION
**Pull request by Beedle-cmyk**

### CODE REVIEW

File: linkedlists/linkedlist_class.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace linkedlist
+{
+	class List
+	{
+	private:
+
+		struct node
+		{
+			int data;
+			node* next;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h:11]: Get rid of excess/unecessary white space

File: linkedlists/linkedlist_class.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+		};
+
+
+		node* head;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: Use 'p' prefix for point types e.g. pHead

File: linkedlists/linkedlist_class.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+//Creation of a class called List with namespace linkedlist
+namespace linkedlist
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: Use something like 'Beetle-cmyk.InterviewQuestions' for your namespace

File: linkedlists/linkedlist_class.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+{
+	class List
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: class name: LinkedList

File: linkedlists/linkedlist_class.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+		struct node
+		{
+			int data;
+			node* next;
+
+		};
+
+
+		node* head;
+		node* curr;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: why are curr and temp class fields instead of local variables?

File: linkedlists/linkedlist_class.h [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+		node* temp;
+
+	public:
+
+		List();
+		void AddNode(int addData);
+		void DeleteNode(int delData);
+		void PrintList();
+	};
+
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: ws

File: linkedlists/linkedlist_class.h [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+		};
+
+
+		node* head;
+		node* curr;
+		node* temp;
+
+	public:
+
+		List();
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: Where is ~List()? (the class destructor?)

File: linkedlists/linkedlists.cpp [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+    node* n = new node;
+    n->next = NULL;
+    n->data = addData;
+
+    //Checking to see if there is already an item/data in the list
+    //if so advance through the list until a NULL/ BLANK NODE is reached and replace that node with the new one
+    //else if the node is already blank than the start is now the added node
+    if (head != NULL)
+    {
+        curr = head;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp]: use local variable for pCurr

File: linkedlists/linkedlists.cpp [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    else
+    {
+        head = n;
+    }
+}
+
+void List::DeleteNode(int delData) 
+{
+    node* delPtr = NULL;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp:56]: use 'pDel'

File: linkedlists/linkedlists.cpp [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+    temp = head;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp]: use local pTemp

File: linkedlists/linkedlists.cpp [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    curr = head;
+    while (curr != NULL and curr->data != delData)
+    {
+        temp = curr;
+        curr = curr->next;
+    }
+    if (curr == NULL)
+    {
+        std::cout << delData << "Was not in the list" << std::endl;
+        delete delPtr;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp:67]: add a test for this conditional; it looks like delPtr is NULL, which should cause an access violation/segmentation fault. Read about operating system memory managers to find out why this crashes here.

File: linkedlists/linkedlists.cpp [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+    }
+    else
+    {
+        delPtr = curr;
+        curr = curr->next;
+        temp->next = curr;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp]: how is temp used?

File: linkedlists/linkedlists.cpp [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+        if (delPtr == head)
+        {
+            head = head->next;
+            temp = NULL;
+        }
+        delete delPtr;
+        std::cout << "The value " << delData << "was deleted" << std::endl;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp:80]: what if there is more than one value matching delData?

File: linkedlists/linkedlist_class.h [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+		void AddNode(int addData);
+		void DeleteNode(int delData);
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h:19]: Define and implement void Clear();

File: linkedlists/linkedlists.cpp [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+        std::cout << curr->data << std::endl;
+        curr = curr->next;
+    }
+}
+
+
+
+int main()
+{
+    List Me;
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlists.cpp]: Instead of a stack object, use a heap object. E.g.:
LinkedList* pTestList = new LinkedList();
This will help you test your new destructor once the code calls delete pTestList on line 108.

File: linkedlists/linkedlist_class.h [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+		void PrintList();
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h:20]: Define and implement ~LinkedList (destructor)

File: linkedlists/linkedlist_class.h [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+	};
+	
+
```
Comment (Beedle-cmyk) [inline linkedlists/linkedlist_class.h]: define and implement void Reverse()

File: linkedlists/linkedlist_class.h (hunk 16 above)
Comment (Beedle-cmyk) [reply linkedlists/linkedlist_class.h:20]: Also implement int Count()
